### ISSUE TITLE
Ctrl + Esc modifier sequences don't send anything

### ISSUE THREAD

**Issue by treyharris** [blinksh/blink#263]
I'm not sure if this is the same as #117 or not; given the examples, it's different, but it may arise from the same source.

In Emacs, many navigation commands are synthetic: _`C-<key>`_  does character navigation, _`M-<key>`_ does line navigation, and so _`C-M-<key>`_ does lexical navigation. But Blink doesn't appear to send anything at all when both the keys mapped to Ctrl and mapped to Esc are held simultaneously when another key is pressed.

I'm left with having to manually hit _`Esc C-<key>`_, which is painful for this since, as navigation commands, they're meant to be used repeatedly.

I don't see a technical issue with why Blink can't send Esc + Ctrl?

_Update: I just discovered by trial and error that with Ctrl set to Ctrl and Alt set to Esc,_  `C-M-<x>`, _where _x_ is a digit, *does* send something: it sends the digit alone without the escape. Of course, Ctrl-[0-9] are undefined in ASCII and in Unicode, so sending them via Blink is impossible. But it is a divergence from the behavior with keys for which Ctrl variants are available, so it might be diagnostically useful._

**Comment by carloscabanero** [comment]
Good catch, it is definitely related to your last comment. Will define it as there is no reason why not to, specially when alt is esc.

**Comment by jra** [comment]
Also in emacs, with caps lock->control and alt->esc, M-$ (alt-shift-4) doesn't get you spellcheck as expected, but an Eval prompt instead. Though, M-{ and M-} (alt-shift-[ and alt-shift-]) do work as expected in emacs---jump to the next empty line.

Also, C-% (ctrl-shift-5) and M-C-% (alt-ctrl-shift-5) don't work as expected in emacs---search and replace, and with reflex).

I'm not sure what his means in terms of codes being sent, but I hope it helpful.

**Comment by carloscabanero** [comment]
Will try to fit this in the next small release, before we do a breakthrough one :)

**Comment by treyharris** [comment]
@jra Just FYI: `C-%` is not a Unicode character sequence, so Blink will never be able to send it. You can easily check as to whether a key sequence is a character or not in Emacs. Go to your `*scratch*` buffer, and enter the following Elisp:

```elisp
(kbd "C-%")
```

Then, after the close-paren, press `C-x C-e` to evaluate. If it's something Blink can even _possibly_ send, it will return a string, usually something like `"^E"`. Otherwise, it will return a vector (which will be represented as a vector containing a large integer).

Note that this only works for _single keystrokes_, and that Meta is not actually a "modifier" in TTY Emacs, it's just a way to say "the escape key comes before this". So when you press alt-0 in Blink, Blink sends Escape, then 0, in quick succession. So the above technique will only work for single keystrokes containing just "C-<something>".

**Comment by treyharris** [comment]
@jra And a second FYI: There's a host of commands Emacs provides when you absolutely _must_ send one of those characters:

```
C-x @           Prefix Command                                                                                                                                            
                                                                                                                                                                          
C-x @ S         event-apply-shift-modifier                                                                                                                                
C-x @ a         event-apply-alt-modifier                                                                                                                                  
C-x @ c         event-apply-control-modifier                                                                                                                              
C-x @ h         event-apply-hyper-modifier                                                                                                                                
C-x @ m         event-apply-meta-modifier                                                                                                                                 
C-x @ s         event-apply-super-modifier                                                                                                                                
```

So, you could, if you had to, press `C-x @ c %`. It's really better to rebind your keys to something you can produce, though. 😛
